### ISSUE TITLE
`hook-` does not recompute key index

### ISSUE THREAD

**Issue by nasser** [arcadia-unity/Arcadia#186]
`hook-` calls `ArcadiaBehaviour.RemoveFunction` which is not implemented correctly. It needs to recompute `fnIndexes` otherwise keys become incorrect after the first removal.

**Comment by selfsame** [comment]
simple repro
```clj
(let [o (UnityEngine.GameObject. "o")]
  (hook+ o :update :a identity)
  (hook+ o :update :b #(log %))
  (hook- o :update :a)
  (hook- o :update :b))
```

**Comment by nasser** [comment]
Should be fixed by #222
